core/crdt/pncounter.go

```diff
@@ -108,13 +114,30 @@ func (reg PNCounter[T]) Value(ctx context.Context) ([]byte, error) {
 }
 
 // Set generates a new delta with the supplied value
-func (reg PNCounter[T]) Increment(value T) *PNCounterDelta[T] {
+func (reg PNCounter[T]) Increment(ctx context.Context, value T) (*PNCounterDelta[T], error) {
+	// To ensure that the dag block is unique, we add a random number to the delta.
+	// This is done only on update (if the doc doesn't already exist) to ensure that the
+	// initial dag block of a document can be reproducible.
```

Comment:
praise: Good catch :)

question: Do we have any tests that target this and ensure that the docID is consistent (e.g. by hard coding it in the test)? If not, please consider this a todo :)

Reply:
Should have read first, there are tests that cover this, but they are complicated by cids and updates.  Can you please add a simple one that makes it clear why such a test is needed?

Reply:
If the docID wasn't consistent, the updates wouldn't sync over P2P.

Reply:
True :)  Thanks for pointing out

Reply:
I'm adding a docID assertion on the mutation create test just to make it more obvious.